### PR TITLE
Asterisk 16

### PR DESCRIPTION
**Pull request by jkroonza**

### CODE REVIEW

File: net-misc/asterisk-g729/asterisk-g729-16.0.3.1.9.ebuild [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+inherit multilib
```
Comment (juippis) [inline net-misc/asterisk-g729/asterisk-g729-16.0.3.1.9.ebuild:6]: Hmm, does this use multilib at all?

File: net-misc/asterisk-g729/asterisk-g729-16.0.3.1.9.ebuild [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+src_install() {
+	local binsuffix
+
+	if use x86; then
+		binsuffix=x86_32
+	elif use amd64; then
+		binsuffix=x86_64
+	else
+		die "Ebuild only functions for x86 and amd64."
+	fi
```
Comment (juippis) [inline net-misc/asterisk-g729/asterisk-g729-16.0.3.1.9.ebuild:59]: Your KEYWORDS and SRC_URI make it clear already.

File: net-misc/asterisk/asterisk-16.9.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+	# Voicemail storage ...
+	for vmst in ${IUSE_VOICEMAIL_STORAGE/+/}; do
+		if use ${vmst}; then
+			menuselect --enable $(echo ${vmst##*_} | tr '[:lower:]' '[:upper:]')_STORAGE menuselect.makeopts
+		fi
+	done
+
+	if use debug; then
+		for o in DONT_OPTIMIZE DEBUG_THREADS BETTER_BACKTRACES; do
+			menuselect --enable $o menuselect.makeopts
```
Comment (juippis) [inline net-misc/asterisk/asterisk-16.9.0.ebuild:228]: `${o}`

File: net-misc/asterisk/asterisk-16.9.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+	elog "http://www.voip-info.org/wiki-Asterisk"
+	echo
+	elog "http://www.automated.it/guidetoasterisk.htm"
+	echo
+	elog "Gentoo VoIP IRC Channel:"
+	elog "#gentoo-voip @ irc.freenode.net"
+	echo
+	echo
+	elog "Please read the Asterisk $(ver_cut 1) upgrade document:"
+	elog "https://wiki.asterisk.org/wiki/display/AST/Upgrading+to+Asterisk+$(ver_cut 1)"
```
Comment (juippis) [inline net-misc/asterisk/asterisk-16.9.0.ebuild:330]: So I saw this while testing the initd update. There was just a discussion in gentoo-dev mailing list that these kind of messages pollute the important output when doing world updates etc.

I'd honestly cut this all to a simple link to gentoo wiki where you can have all this information. "Asterisk has been installed" is also the default expectation if you see these messages...

Or make it so this is printed only the first time asterisk is installed, not in every update.

File: net-misc/asterisk-g729/asterisk-g729-16.0.3.1.9.ebuild (hunk 1 above)
Comment (jkroonza) [reply net-misc/asterisk-g729/asterisk-g729-16.0.3.1.9.ebuild:6]: $(get_libdir) previously required inherit multilib.  Removed.

File: net-misc/asterisk-g729/asterisk-g729-16.0.3.1.9.ebuild (hunk 2 above)
Comment (jkroonza) [reply net-misc/asterisk-g729/asterisk-g729-16.0.3.1.9.ebuild:59]: I don't like accidental fall-throughs.  Removed anyway.

File: net-misc/asterisk/asterisk-16.9.0.ebuild (hunk 4 above)
Comment (jkroonza) [reply net-misc/asterisk/asterisk-16.9.0.ebuild:330]: I agree.  Down to three lines.

**Comment by jkroonza** [comment]
> ```
>  * ERROR: net-misc/asterisk-16.9.0::testworld failed (install phase):
>  *   !!! newinitd: /tmp/portage/net-misc/asterisk-16.9.0/files/initd-13.32.0-r1 does not exist

Yea.  That would.  I screwed up the rebase post the security push.  rebased onto newest master again now, seems to be happier on my side at least.  Think when I tested the file was actually still remnant as an artifact of the rebase.

>  * QA Notice: Unrecognized configure options:
>  * 
>  * 	--without-janson-bundled
>  * 	--without-janson-bundled
> ```

Bugger.  I keep misspelling that.  I really need to get that QA stuff working properly though.  Should also be fixed now.

**Pull request closed without merging**